index.js

```diff
@@ -78,8 +78,9 @@ class KeyringController extends EventEmitter {
    * @returns {Promise<Object>} A Promise that resolves to the state.
    */
   async createNewVaultAndKeychain(password) {
-    await this.createFirstKeyTree(password);
-    await this.persistAllKeyrings(password);
+    this.password = password;
+
+    await this.createFirstKeyTree();
```

Comment:
`createFirstKeyTree` only modifies `this.keyrings` via `addNewKeyring`, which calls `persistAllKeyrings` internally after adding the keyring. That's why the `persistAllKeyrings` call was eliminated here.